### DISCUSSION TITLE
How can I add a Vehicle Selection tab?

### DISCUSSION THREAD

**Discussion by KaoZ21** [community/community#121107] [Programming Help]
### Select Topic Area

Question

### Body

Currently, we are designing a drone and a land vehicle as a team at my school. I wrote the following code to capture drone information, but I need to add it in a tab here to change the vehicle. What should I do?

```
from tkinter import *
from pymavlink import mavutil
import datetime
from PIL import Image, ImageTk

address = 'udpin:localhost:14550'
# address = '/dev/ttyACM0'
# address = '/dev/ttyTHS1:115200'

vehicle = mavutil.mavlink_connection(address, baud=57600, autoreconnect=True)
vehicle.wait_heartbeat()
print('Bağlantı Başarılı')

root = Tk()
root.geometry("1280x720")
root.configure(bg = "#ffffff")

canvas = Canvas(
    root,
    bg = "#ffffff",
    height = 1080,
    width = 1920,
    bd = 0,
    highlightthickness = 0,
    relief = "ridge")
canvas.place(x = 0, y = 0)

background_img = PhotoImage(file = f"/home/kara/Templates/kara/background.png")
background = canvas.create_image(
    640.0, 370.0,
    image=background_img)

battery_label = Label(root, text="Batarya Yüzdesi: -", font=("Arial", 25), background='#D9D9D9')
battery_label.place(x=40, y=57)

lat_lng_label = Label(root, text="LAT: -\nLNG: -", font=("Arial", 25),background='#D9D9D9')
lat_lng_label.place(x=1252, y=57, anchor=NE)

altitude_label = Label(root, text="LNG: -", font=("Arial", 25), background='#D9D9D9')
altitude_label.place(x=1242, y=140, anchor=NE)

mode_label = Label(root, text="Uçuş Modu: -", font=("Arial", 25),background='#D9D9D9')
mode_label.place(x=25, y=665)

speed_label = Label(root, text="Hız: -", font=("Arial", 25),background='#D9D9D9')
speed_label.place(x=60, y=133)

pitch_label = Label(root, text="Pitch: -", font=("Arial", 25),background='#D9D9D9')
pitch_label.place(x=72, y=463)

roll_label = Label(root, text="Roll: -", font=("Arial", 25), background='#D9D9D9')
roll_label.place(x=75, y=393)

yaw_label = Label(root, text="Yaw: -", font=("Arial", 25), background='#D9D9D9')
yaw_label.place(x=1169, y=394, anchor=NE)


def update_info():
    msg = vehicle.recv_match(type='BATTERY_STATUS', blocking=True)
    battery_label.config(text=f'Batarya Yüzdesi: %{msg.battery_remaining}')

    msg2 = vehicle.recv_match(type='SIMSTATE', blocking=True)
    lat_lng_label.config(text=f'LAT: {msg2.lat}')

    msg4 = vehicle.recv_match(type='SIMSTATE', blocking=True)
    altitude_label.config(text=f'LNG: {msg4.lng}')

    msg3 = vehicle.recv_match(type='HEARTBEAT', blocking=True)
    mode = mavutil.mode_string_v10(msg3)
    mode_label.config(text=f'Uçuş Modu: {mode}')

    msg_speed = vehicle.recv_match(type='GPS_RAW_INT', blocking=True)
    speed_label.config(text=f'Hız: {msg_speed.vel} m/s')

    msg5 = vehicle.recv_match(type='AHRS2', blocking=True)
    pitch = str(msg5.pitch)[:4]  # İlk 3 haneyi alır
    pitch_label.config(text=f'Pitch: {pitch}')

    msg6 = vehicle.recv_match(type='AHRS2', blocking=True)
    roll = str(msg6.roll)[:4]
    roll_label.config(text=f'Roll: {pitch}')

    msg7 = vehicle.recv_match(type='AHRS2', blocking=True)
    yaw = str(msg7.yaw)[:3]
    yaw_label.config(text=f'Yaw: {yaw}')

    today = datetime.date.today()
    date_label.config(text=today.strftime('%d-%m-%Y'), background='#D9D9D9', font=('Arial', 25))

    root.after(2000, update_info)  # Her 2 saniyede verileri yeniler

def arm_disarm():
    vehicle.arducopter_arm()
    armed_label.config(text="Arm Durumu Değiştirildi", font=('Arial', 25))

arm_disarm_button = Button(root, text="Arm/Disarm", command=arm_disarm, background='#D9D9D9')
arm_disarm_button.place(x=30, y=593)

armed_label = Label(root, text="Armed/Disarmed: -", font=("Arial", 25), background='#D9D9D9')
armed_label.place(x=150, y=590)

date_label = Label(root, text="", font=("Arial", 20),background='white')
date_label.place(x=1230, y=664, anchor=NE)

update_info()
root.mainloop()
`

```

**Comment by ZFPuhi** [comment]
Hello,

You can use the ttk module from Tkinter to create tabs easily. Here is a sample look:

`
from tkinter import *
from tkinter import ttk
from pymavlink import mavutil
import datetime
from PIL import Image, ImageTk

def capture_drone_info():
    # Your code to capture drone information goes here
    pass

def capture_vehicle_info():
    # Your code to capture land vehicle information goes here
    pass

def switch_to_drone():
    capture_drone_info()

def switch_to_vehicle():
    capture_vehicle_info()

root = Tk()
root.geometry("1280x720")
root.configure(bg="#ffffff")

notebook = ttk.Notebook(root)
notebook.pack(fill=BOTH, expand=True)

# Drone Tab
drone_tab = Frame(notebook)
notebook.add(drone_tab, text='Drone')

drone_canvas = Canvas(
    drone_tab,
    bg="#ffffff",
    height=1080,
    width=1920,
    bd=0,
    highlightthickness=0,
    relief="ridge")
drone_canvas.pack()

drone_background_img = PhotoImage(file="background.png")
drone_background = drone_canvas.create_image(
    640.0, 370.0,
    image=drone_background_img)

drone_switch_button = Button(drone_tab, text="Switch to Vehicle", command=switch_to_vehicle)
drone_switch_button.pack()

# Vehicle Tab
vehicle_tab = Frame(notebook)
notebook.add(vehicle_tab, text='Vehicle')

vehicle_canvas = Canvas(
    vehicle_tab,
    bg="#ffffff",
    height=1080,
    width=1920,
    bd=0,
    highlightthickness=0,
    relief="ridge")
vehicle_canvas.pack()

vehicle_background_img = PhotoImage(file="background.png")
vehicle_background = vehicle_canvas.create_image(
    640.0, 370.0,
    image=vehicle_background_img)

vehicle_switch_button = Button(vehicle_tab, text="Switch to Drone", command=switch_to_drone)
vehicle_switch_button.pack()

root.mainloop()
`

As you can see in the example above I have used the 'ttk.Notebook' to create tabs, one is the 'drone_tab' and then the second one with the land vehicle information 'vehicle_tab'. Each of the tabs have their own canvas and switch button to make it more easy and simple.

**Reply by KaoZ21** [reply]
Thanks!!

**Reply by ZFPuhi** [reply]
Hope I was able to help and resolve your question ^_^

**Reply by ebndev** [reply]
Hey @KaoZ21! If you found this answer helpful, please feel encouraged to Mark it as answered 😁

The project sounds really cool btw!

**Comment by LiteBrite82** [comment]
Thanks for posting in the GitHub Community, @KaoZ21 !

We're happy you're here. You are more likely to get a useful response if you are posting your question in the applicable category, the Code Search and Navigation category is solely related to conversations around the [GitHub product Code Search and Navigation](https://github.com/search?q=&type=repositories). This question should be in the Programming Help category. I've gone ahead and moved it for you. Good luck!
